op-batcher/batcher/channel_manager_test.go

```diff
@@ -106,24 +108,193 @@ func TestChannelManager_TxResend(t *testing.T) {
 
 	txdata0, err := m.TxData(eth.BlockID{})
 	require.NoError(err)
-	txdata0bytes := txdata0.Bytes()
-	data0 := make([]byte, len(txdata0bytes))
+
+	// confirm one frame to keep the channel open
+	m.TxConfirmed(txdata0.ID(), eth.BlockID{})
+
+	txdata1, err := m.TxData(eth.BlockID{})
+	require.NoError(err)
+	txdata1bytes := txdata1.Bytes()
+	data1 := make([]byte, len(txdata1bytes))
 	// make sure we have a clone for later comparison
-	copy(data0, txdata0bytes)
+	copy(data1, txdata1bytes)
 
 	// ensure channel is drained
 	_, err = m.TxData(eth.BlockID{})
 	require.ErrorIs(err, io.EOF)
 
 	// requeue frame
-	m.TxFailed(txdata0.ID())
+	m.TxFailed(txdata1.ID())
 
-	txdata1, err := m.TxData(eth.BlockID{})
+	txdata2, err := m.TxData(eth.BlockID{})
 	require.NoError(err)
 
-	data1 := txdata1.Bytes()
-	require.Equal(data1, data0)
-	fs, err := derive.ParseFrames(data1)
+	data2 := txdata2.Bytes()
+	require.Equal(data2, data1)
+	fs, err := derive.ParseFrames(data2)
 	require.NoError(err)
 	require.Len(fs, 1)
 }
+
+// TestChannelManagerCloseBeforeFirstUse ensures that the channel manager
+// will not produce any frames if closed immediately.
+func TestChannelManagerCloseBeforeFirstUse(t *testing.T) {
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	log := testlog.Logger(t, log.LvlCrit)
+	m := batcher.NewChannelManager(log, batcher.ChannelConfig{
+		TargetFrameSize:  0,
+		MaxFrameSize:     100,
+		ApproxComprRatio: 1.0,
+		ChannelTimeout:   1000,
+	})
+
+	a, _ := derivetest.RandomL2Block(rng, 4)
+
+	err := m.Close()
+	require.NoError(t, err)
+
+	err = m.AddL2Block(a)
+	require.NoError(t, err)
+
+	_, err = m.TxData(eth.BlockID{})
+	require.ErrorIs(t, err, io.EOF)
+}
+
+// TestChannelManagerCloseNoPendingChannel ensures that the channel manager
+// can gracefully close with no pending channels, and will not emit any new
+// channel frames.
+func TestChannelManagerCloseNoPendingChannel(t *testing.T) {
+	log := testlog.Logger(t, log.LvlCrit)
+	m := batcher.NewChannelManager(log, batcher.ChannelConfig{
+		TargetFrameSize:  0,
+		MaxFrameSize:     100,
+		ApproxComprRatio: 1.0,
+		ChannelTimeout:   1000,
+	})
+	lBlock := types.NewBlock(&types.Header{
+		BaseFee:    big.NewInt(10),
+		Difficulty: common.Big0,
+		Number:     big.NewInt(100),
+	}, nil, nil, nil, trie.NewStackTrie(nil))
+	l1InfoTx, err := derive.L1InfoDeposit(0, lBlock, eth.SystemConfig{}, false)
+	require.NoError(t, err)
+	txs := []*types.Transaction{types.NewTx(l1InfoTx)}
+
+	a := types.NewBlock(&types.Header{
+		Number: big.NewInt(0),
+	}, txs, nil, nil, trie.NewStackTrie(nil))
```

Comment:
We'll soon merge https://github.com/ethereum-optimism/optimism/pull/5167 which contains a function `newMiniL2Block` removing this duplicated code.